### ISSUE TITLE
Cinnamon crashes on every boot since Mint 19

### ISSUE THREAD

**Issue by MXarko** [linuxmint/muffin#413]
```
 * muffin version (muffin --version or 'dpkg --list | grep libmuffin0' for Mint/Ubuntu)
muffin 4.0.6

 * Distribution - (Mint 17.2, Arch, Fedora 25, etc...)
Linux Mint 19.1 Tessa

 * Graphics hardware *and* driver used
Device-1: AMD RS690 [Radeon X1200] vendor: Micro-Star MSI K9AG Neo2
driver: radeon v: kernel bus ID: 01:05.0 chip ID: 1002:791e
Display: x11 server: X.Org 1.19.6 driver: ati,radeon
unloaded: fbdev,modesetting,vesa resolution: 1680x1050~60Hz
OpenGL: renderer: ATI RS690 v: 2.1 Mesa 18.0.5 direct render: Yes

 * 32 or 64 bit
64 bit

 ```

**Issue**
Linux Mint 19 and 19.1 Cinnamon on every boot right after login give me pop-up “Cinnamon just crashed…”. 18.3 and previous versions work fine.

I installed Debian 9.6 Cinnamon on the same PC, and there was a similar problem: right after login there was a notification that Cinnamon is running without video hardware acceleration, etc. So I looked for a package that would support my video card, Radeon RS690. And I found it in Debian repository, "firmware-amd-graphics (20161130-4) [non-free]". I installed it (in Debian), and it solved the problem completely.

So I had thought that the same will work in Mint. But when I tried to install the same package in Mint, I've got the following error message:

Unpacking firmware-amd-graphics (20161130-4) ...
dpkg: error processing archive firmware-amd-graphics_20161130-4_all.deb (--install):
 trying to overwrite '/lib/firmware/amdgpu/carrizo_ce.bin', which is also in package linux-firmware 1.173.2
dpkg-deb: error: paste subprocess was killed by signal (Broken pipe)

Then I uninstalled "linux-firmware" package, and three more that depend on it, in order to install the AMD one. I succeeded at installing "firmware-amd-graphics" package, rebooted, but got the same Cinnamon crash pop-up. I installed "firmware-linux-nonfree" and "firmware-misc-nonfree" as well (both from the Debian site), but the result was the same.

I also applied the file attached in issue https://github.com/linuxmint/Cinnamon/issues/8185 , "muffin-4.0.5-cogl-rect-new-compat.tar.gz", but no joy.

Then I rolled everything back and continued to use Mint in Fallback Mode. Even so it has a few more unpleasant, to say the least, symptoms:
1. Every time I try to login at Yahoo mail, using Firefox, by navigating to "mail.yahoo.com" and entering my e-mail address, I get "the tab just crashed" page. I get the same crash page on some other sites. (There is not that problem in Google Chrome.)
2. I can not plot anything in Octave. I tried all of the three available graphics toolkits (fltk, gnuplot, qt), and when I issue, for example, "plot(3,2)" command Octave crashes (or freezes).

**Other information**
Mntreport stack trace (made with debug symbols (muffin-dbg, cinnamon-dbg) installed) :
https://termbin.com/03wn

dmesg:
[dmesg.txt](https://github.com/linuxmint/muffin/files/2758842/dmesg.txt)

$ inxi -F
System:
Host: ***** Kernel: 4.15.0-43-generic x86_64 bits: 64
Desktop: Cinnamon 4.0.9 Distro: Linux Mint 19.1 Tessa
Machine:
Type: Desktop Mobo: MICRO-STAR model: MS-7367 v: 1.0
serial: BIOS: American Megatrends v: 3.9 date: 01/21/2010
CPU:
Topology: Dual Core model: AMD Athlon 64 X2 5200+ bits: 64 type: MCP
L2 cache: 1024 KiB
Speed: 1800 MHz min/max: 1000/2700 MHz Core speeds (MHz): 1: 1800 2: 1800

~/.xsession-errors:
[xsession-errors.txt](https://github.com/linuxmint/muffin/files/2758847/xsession-errors.txt)

**Comment by jaszhix** [comment]
Are you using any drivers or firmware meant for Debian in Mint? If so, I would remove those. Maybe see if a live session from disk works. If so, then there's probably something wrong with the drivers.

There are no symbols in the trace provided, so no way of figuring out where in cinnamon or muffin this is crashing.

**Comment by MXarko** [comment]
No, installation is pretty clean, I have made only the aforementioned changes, but undid them. I took care not to change anything important, for the purpose of troubleshooting, because I had the issue in the earlier installations of the same Mint Cinnamon version. Live session (from USB flash) has the same behaviour, Cinnamon crashes.

What should I do regarding missing symbols? I have earlier turned Debug symbols in Software Sources on and confirmed that, and checked that the two related packages have been installed.

**Comment by jaszhix** [comment]
Not sure, if symbols are installed, then that would suggest wherever it's crashing in the trace is not muffin - could be in mesa or the kernel, in which case there's not much I can do. Usually it shows where the call originated from if that is the case, but doesn't here.

There's a few settings you could try playing with to see if one of them works better with your driver.

`gsettings set org.cinnamon.muffin threaded-swap false`
`gsettings set org.cinnamon.muffin send-frame-timings false`
`gsettings set org.cinnamon.muffin sync-to-vblank false`

Those are new as of 4.0. You said it's also crashing in Mint 19 (cinnamon 3.8), so those settings may not help, but worth testing.

**Comment by MXarko** [comment]
I suspect mesa is the culprit, because in my earlier investigation I updated everything but mesa and there were no (obvious) problems, but when I updated mesa the problem I investigated came back. Maybe it was not the same problem as this one, it was couple of months ago, but there is something fishy about the new mesa in my system.

What can I try in order to solve it? I will try your suggestion, but what if it fails?

**Comment by jaszhix** [comment]
You could try the settings I suggested above in a terminal - you can toggle them with true or false at the end of the command. Try them one at a time, and try to start cinnamon.
 
For mesa, you could try one of the PPAs that have newer versions. 18.2.x might have better mileage, but at the same time might cause regressions, or not work at all with a chip that old.

**Comment by MXarko** [comment]
Thank you for your patience! I installed a new dedicated graphics card (and apropriate driver) and now everything works well. Problems with Firefox crashing at Yahoo mail and Octave crashing when plotting have gone as well.

So, we can conclude that Linux Mint Cinnamon from version 19 onwards does not support (at least) Radeon RS690 graphics.

**Comment by Redsandro** [comment]
This looks similar to issue #8592, except for me 19.0 works fine but 19.1 does not. Using nvidia GeForce GTX 970.

**Comment by bonaldi69** [comment]
Take a look at[ https://github.com/linuxmint/cinnamon/issues/8813]( https://github.com/linuxmint/cinnamon/issues/8813), I think it's complementary.
